### PR TITLE
flycheck-coala.el: json -> format output

### PR DESCRIPTION
**Pull request by fleimgruber**
The ``--format`` option is more robust than the ``--json`` option when
logging output is mixed with the output of ``:command``;
``json-read-from-string`` breaks in this case.

Fixes https://github.com/coala/coala-emacs/issues/15

### CODE REVIEW

File: flycheck-coala.el [hunk 1]
```diff
@@ -30,42 +30,14 @@
...
-                         (cdr (assoc 'severity err)))) errors))))
-    errors))
 
 (flycheck-define-checker coala
   "A checker using coala.
 See URL `https://coala.io'."
-  :command ("coala" "--json" "--find-config" "--files" source)
-  :error-parser flycheck-coala-parse-json
+  :command ("coala" "--format" "{line}:{column}:{severity}:{message}" "--find-config" "--files" source)
+  :error-patterns  ((error (or "None" line) ":" (or "None" column) ":" (any "03") ":" (message))
```
Comment (sils) [inline flycheck-coala.el]: why 03 and not 3?

Comment (sils) [reply flycheck-coala.el]: if 03 for a reason this needs a comment IMO

Comment (userzimmermann) [inline flycheck-coala.el]: Can there really be severity `0`?

File: flycheck-coala.el [hunk 2]
```diff
@@ -30,42 +30,14 @@
...
+                    (warning (or "None" line) ":" (or "None" column) ":" (any "2") ":" (message))
```
Comment (userzimmermann) [inline flycheck-coala.el]: `":2:"` should be enough instead of using `(any "...")`

**Review by userzimmermann — CHANGES_REQUESTED** [review]
Hey @fleimgruber :) Thanks for the PR! Always great to see other Emacs enthusiasts in the coala community... I am only wondering about some little things ;)

**Comment by fleimgruber** [comment]
Thanks for your comments!  I changed this to use `severity_str` and literal matches since these map nicely to `flycheck` error levels.  I also added tests to this, see #11.  WDYT?

File: flycheck-coala.el (hunk 1 above)
Comment (fleimgruber) [reply flycheck-coala.el]: I do not use numerical error codes any more, see below.

Comment (fleimgruber) [reply flycheck-coala.el]: I do not use numerical error codes any more, see below.

File: flycheck-coala.el (hunk 2 above)
Comment (fleimgruber) [reply flycheck-coala.el]: dito

**Review by userzimmermann — APPROVED** [review]

**Comment by userzimmermann** [comment]
ack https://github.com/coala/coala-emacs/pull/16/commits/a9253f192403f067c9a62b47897fb51b4a9a3903

**Comment by userzimmermann** [comment]
@rultor merge

**Comment by userzimmermann** [comment]
@fleimgruber Now create the PR for https://github.com/coala/coala-emacs/issues/11 :)

**Pull request merged**